README.MD
# wifidisplay8x8

Bandeau led pilotable en wifi, avec mode wifi AP pour configuration.

<h2>installation chez moi</h2>
<img src="led_8x8.JPG" width="25%" height="25%"/>

<h2>interface d'administration</h2>
<img src="interface.PNG" width="25%" height="25%"/>


utilise esp8266/Arduino : https://github.com/esp8266/Arduino

Basé sur le code pour l'affichage de 
   https://github.com/SensorsIot/MAX7219-4-digit-display-Library-for-ESP8266-/tree/master/MAX7219_Dot_MatrixESP-master
et pour l'auto config de 
   https://github.com/Orkeren/ESP8266-Arduino-Webconfig



CHANGES
* 2017/08/15
  - correction BUG 08131 
  - correction BUG 08132
  - Si config existe essaye de se connecté en 1er au wifi sauvegardé
  - Affichage de message d'attente quand il tente de se connecter au wifi
  - Mise en place de valeur de vitesse plus cohérente, et initialisation de la page message ok pour la vitesse.
  - EVOL 081501 : Gestion du message par défaut avec un bouton ou selon l'édition du premier caractere pour efface le message ou editer, a voir sur un site web 

* 2017/08/13
  - Simplification du code sur la gestion des message initiaux.
  - BUG 08131 : Le message initial ne s'affiche pas avant la connection au wifi si celui ci est le bon
  - BUG 08132 : Impossible de se connecter en HTTP au module !!!!!
  
* 2017/06/11
 - LE bug A2 est corrigé, mais j'ai pas compris
 - http://a.b.c.d redirige soit sur admin.html en mode admin ou message.html dans l'autre cas.
 - affiche un info bulle pour expliquer les variable sur le champe "Message a afficher"
 
* 2017/06/10
 - Mise en Place de la Variable NoFont, en tant que variable de la classe MAX7219_Dot_Matrix
 - Correction de la gestion du nombre de LEd 8x8 (BUG A1) en créeant une fonction reinit qui nécessite tout de même de reseter le controleur pour que cela fonctionne
 -=- BUG A2 : La gestion de l'offset négatif dans sendSmooth ne fonctionne pas correctement, rien de n'affiche pendant longtemps


* 2017/06/06
  - Mise en place de la gestion des variable pour heure (<hh:mm:ss>) et date (<jj/mm/aa> et <jj/mm>)
  - Mise en place de la gestion du nombre de MAX7219 dans la tructure config et EEPROM,
  -=- BUG A1: le nombre MAX7219 est bien sauvegardé, mais n'a pas l'aire d'etre pris en compte 
        Peut etre du au faite de l'initialiser 2 fois pour que ce soit une variable globale
  - TODO TD1: Ajouté a la page principale un bulle d'aide pour les variables et a terme les caractere spéciaux 
        ex : Train, température, pluie, soleil, .......


* 2017/06/05
  - Creation de la fct de recuperation des valeurs actuel pour message
  - Amélioration de l'affichage de la page de saisie de message
  
* 2017/06/04
 - Fin de la gestion de config avec mise en place d'un CSS dédier et d'un fichier ico associé
 - Creation du fichier PAge_Message.h qui reprend les fonction qui existait dans .ino
 TODO : 
    Regrouper toutes les HEADER et FOOTER des pages

* 2017/06/02
 - Ajout de la gestion de configuration en mode AP d'après :
   https://www.john-lassen.de/en/projects/esp-8266-arduino-ide-webconfig

* 2017/05/28
 - Ajout de 2 module suplémentaire.
 - Pour afficher des police 5 Bit, il faut tout refaire au niveau de l'envoie des donnée
 - Gestion d'un état qui permet d'économiser l'énergie, en arrêtant le Led, mais sans metre encore le controleur dans un etat de sommeil

* 2017/05/26
 - LE problème c'est que tous l'algo est pensé pour envoyé 8 bit * 8bit pour chaque caracetere
 - Quand on en envoie 6 x 6 il faut generer les caracteres suivant avant mais comment ??????

* 2017/05/22
 - ajout d'une font 5 bit de large et tentative d'inserer dans MAX7219_Dot_Matrix::sendSmooth
   -> Les caracteres ne sont pas plus sérré ??????????

* 2017/05/21
 - Ajout de la librairie NTP : https://github.com/arduino-libraries/NTPClient
  -=- Reste a géré le Problème de changement d'heure.
 - LE capteur de presence ne fonctionne toujours pas

* 2017/05/20 
  - Deplacement des fichiers de la librairie "MAX7219_Dot_MatrixESP-master" en local pour la modifier
  - ajout d'une font dans le fichier MAX7219_Dot_Matrix_font
  		 recupéré https://github.com/d2r2/go-max7219/blob/master/fonts.go
  - ajout des lettres àéèêç par aeeec en attendant de reprendre les caractère
  - ajout de la gestions d'une fonte dans "MAX7219_Dot_Matrix.h" et "MAX7219_Dot_Matrix.cpp"
  		Pas comme je le voulais, mais impossible d'avoir un point sur un const !!!!!
  - PRise en compte de la selection de font dans l'interface html 



Page_Message.h
#ifndef PAGE_MESSAGES_H
#define PAGE_MESSAGES_H




//
//   The MESSAGE PAGE
//
const char PAGE_Message[] PROGMEM = R"=====(
<html lang='fr'>
    <head>
        <meta name='viewport' content='width=device-width, initial-scale=1' />
        <meta http-equiv='Content-Type' content='text/html; charset=utf-8' />
        <link href='data:image/x-icon;base64,AAABAAEAEBAQAAEABAAoAQAAFgAAACgAAAAQAAAAIAAAAAEABAAAAAAAgAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAwP/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQABAREQERABAAEBAAAAEAEBAQERABEQARARAQAAEAABAAEBERAREAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' rel='icon' type='image/x-icon' />
        <title> Afficher LED</title>
        <link rel='stylesheet' href='style.css' type='text/css' />
        <script src='microajax.js'></script> 
    </head>
    <body>

      <div id='title'>Message</div>
      <hr>
      <form action='messageSubmit' method='GET'>
        <p><div class="infoBulle" href="">Message à  afficher:
           <span><h3>les variables spéciales</h3>
            <p>&lsaquo;jj/mm/aa&rsaquo; : Affiche la date courante selon le format de la balise</p>
            <p>&lsaquo;hh:mm:ss&rsaquo; : Affiche l'heure courante selon le format de la balise</p>
           </span>
          </div>
        </p>
        <input id='message' type='text' name='message' maxlength="200" size="40"><br>
        
        <p>Vitesse :
        <select  id='vitesse' name='vitesse'>
          <option value="480">escargot</option>
          <option value="240">tortue</option>
          <option value="120">très lent</option>
          <option value="60">lent</option>
          <option value="25">normal</option>
          <option value="15">vite</option>          
          <option value="13">très vite</option>          
          <option value="10">extrème</option>          
          <option value="5">illisible</option>
          
        </select></p>

        <p>Luminositée :
        <select  id='luminosite' name='luminosite'>
          <option value="-1">Eteindre</option>
          <option value="0">0</option>
          <option value="1">1</option>
          <option value="2">2</option>
          <option value="3">3</option>
          <option value="4">4</option>
          <option value="5">5</option>
          <option value="6">6</option>
          <option value="7">7</option>
          <option value="8">8</option>
          <option value="9">9</option>
          <option value="10">10</option>
          <option value="11">11</option>
          <option value="12">12</option>
          <option value="13">13</option>
          <option value="14">14</option>
          <option value="15">15</option>
        </select></p>
        
        <p>Timer présence :
        <select  id='timer_presence' name='timer_presence'>
          <option value="30">30 s</option>
          <option value="60">1 mn </option>
          <option value="180">3 mn</option>
          <option value="300">5 mn</option>
          <option value="600">10 mn</option>
          <option value="900">15mn </option>
          <option value="1800">30 mn</option>
          <option value="3200">1 h</option>
        </select></p>
        
        <p>font :
        <select  id='font' name='font'>
          <option value="0">default</option>
          <option value="1">SINCLAIR_8x8</option>
          <option value="2">font_bold_8x8</option>
          <option value="3">font_6x8</option>
        </select></p><br>
        
        <p>Rotation :
        <select  id='rotation' name='rotation'>
          <option value="0">Normal</option>
          <option value="90">90° droite </option>
          <option value="180">180 <option>
          <option value="300">320</option>
        </select></p>
        
        <input type='submit' value='Submit' style='width:150px' class='btn btn--m btn--blue'><br>
      </form>
      <hr>
      <a href='admin.html' class='bottom'>page d'administration</a>
      <script>                
        window.onload = function (){
        
              setValues('/getCurrentMessage');  //-- this function calls the function on the ESP      
        
        
        }
        function load(e,t,n){if("js"==t){var a=document.createElement("script");a.src=e,a.type="text/javascript",a.async=!1,a.onload=function(){n()},document.getElementsByTagName("head")[0].appendChild(a)}else if("css"==t){var a=document.createElement("link");a.href=e,a.rel="stylesheet",a.type="text/css",a.async=!1,a.onload=function(){n()},document.getElementsByTagName("head")[0].appendChild(a)}}
      </script>
    </body>
  </html>  
  

)=====";
#endif





